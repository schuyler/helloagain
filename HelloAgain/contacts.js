'use strict';

import { Friends } from './models';
let AddressBook = require('react-native-contacts');

export function loadAllContacts() {
    // Load all the contacts from the system addressbook, and within that list,
    // substitute any contacts we already have in our database (and merging any
    // changes from the addressbook). We have to wait until the Contacts model
    // has loaded fully. Returns a Promise that is resolved when everything is
    // loaded.
    return new Promise((resolve, reject) => {
      Friends.hasLoaded.then(() => {
        AddressBook.getAll((err, contacts) => {
          if (err) {
            console.log("Reading addressbook failed:", err);
            reject(err);
            return;
          }
          // Update contacts from the addressbook with whatever we have stored in
          // our database.
          for (var i = 0; i < contacts.length; i++) {
            var storedContact = Friends.load(contacts[i].recordID);
            if (storedContact) {
              Object.assign(storedContact, contacts[i]);
              if (storedContact.isSelected) { // only write back changes to selected contacts
                Friends.save(storedContact);
              }
              contacts[i] = storedContact;
            }
          }
          resolve(contacts);
        });
      });
    });
};
