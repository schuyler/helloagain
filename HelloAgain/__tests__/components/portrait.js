'use strict'

import React from 'react';
import renderer from 'react-test-renderer';

import { CONTACT_FIXTURE } from "../fixtures/contacts"
import Portrait from "../../components/portrait"

it('renders a default', () => {
  expect(renderer.create(
    <Portrait />
  )).toMatchSnapshot();
})


it('renders a uri', () => {
  let fixture = CONTACT_FIXTURE[0]
  expect(renderer.create(
    <Portrait uri="file://../../assets/epic.png" />
  )).toMatchSnapshot();
})
