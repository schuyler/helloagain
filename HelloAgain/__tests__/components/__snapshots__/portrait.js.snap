exports[`test renders a default 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <Image
    source={1} />
</View>
`;

exports[`test renders a uri 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <Image
    source={
      Object {
        "uri": "file://../../assets/epic.png",
      }
    }
    uri="file://../../assets/epic.png" />
</View>
`;
